src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -295,54 +383,141 @@ TaskSketcherElements::TaskSketcherElements(ViewProviderSketch *sketchView)
+        for (int i = 4; i < ui->listMultiFilter->count(); i++) {
+            ui->listMultiFilter->item(i)->setCheckState(item->checkState());
+        }
+    }
+
+    ui->listMultiFilter->blockSignals(false);
+
+    updateVisibility();
+}
+
+void TaskSketcherElements::setItemVisibility(QListWidgetItem* item)
+{
+    /* index
+    0 => Normal
+    1 => Construction
+    2 => External
+    3 => all geos
+    4 => Point
+    5 => Line
+    6 => Circle
+    7 => Ellipse
+    8 => Arc
+    9 => Arc of Ellipse
+    10 => Hyperbola
+    11 => Parabola
+    12 => bspline
+    */
+
+    ElementData* itemData = qvariant_cast<ElementData*>(item->data(Qt::UserRole));
+    bool visibility = true;
+
+    if (ui->filterBox->checkState() == Qt::Unchecked) { item->setHidden(false); return; }
+
+    if (ui->listMultiFilter->item(0)->checkState() == Qt::Unchecked && !itemData->isConstruction) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(1)->checkState() == Qt::Unchecked && itemData->isConstruction) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(2)->checkState() == Qt::Unchecked && itemData->isExternal) { item->setHidden(true); return; }
+
+    if (ui->listMultiFilter->item(4)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomPoint::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(5)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomLineSegment::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(6)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomCircle::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(7)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomEllipse::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(8)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfCircle::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(9)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfEllipse::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(10)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfHyperbola::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(11)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfParabola::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(12)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomBSplineCurve::getClassTypeId()) { item->setHidden(true); return; }
+
+    item->setHidden(false);
+    return;
 }
 
+void TaskSketcherElements::updateVisibility()
+{
+    for (int i = 0; i < ui->listWidgetElements->count(); i++) {
+        setItemVisibility(ui->listWidgetElements->item(i));
+    }
+} 

```

Comment:
Better to code a dedicated function in ElementView and call it.

Reply:
This was original implementation: https://github.com/FreeCAD/FreeCAD/blob/fce72e46c83aea42399c0422205840545f7abb01/src/Mod/Sketcher/Gui/TaskSketcherElements.cpp#L1046

But I can change it if necessary. setItemVisibility and updateVisibility can be transfered to ElementView I guess. But it doesn't really change anything.

Reply:
Moving it makes that `ui->listWidgetElements` doesn't have to be resolved lot of times. ;)

Reply:
Ok I tried to move it, but the problem is that it calls setItemVisibility. setItemVisibility could be changed too, but it needs access to ui->listMultiFilter.